examples/cryptorealtime/src/main/java/source/CryptoMarketTradeUnboundedReader.java

```diff
@@ -0,0 +1,254 @@
+     *
+     * However, this value should be as late as possible. Downstream windows may not be able to close until this watermark passes their end.
+     *
+     * For example, a source may know that the records it reads will be in timestamp order. In this case, the watermark can be the timestamp of the last record read. For a source that does not have natural timestamps, timestamps can be set to the time of reading, in which case the watermark is the current clock time.
+     *
+     * See Window and Trigger for more information on timestamps and watermarks.
+     *
+     * May be called after advance() or start() has returned false, but not before start() has been called.
+     * @return
+     */
+
+    @Override
+    public Instant getWatermark() {
+        return currentTimestamp.minus(new Duration(1));
+    }
+
+
+    /**
+     * Returns a UnboundedSource.CheckpointMark representing the progress of this UnboundedReader.
+     * If this UnboundedReader does not support checkpoints, it may return a CheckpointMark which does nothing, like:
+     *
+     *
+     *  public UnboundedSource.CheckpointMark getCheckpointMark() {
+     *    return new UnboundedSource.CheckpointMark() {
+     *      public void finalizeCheckpoint() throws IOException {
+     *        // nothing to do
+     *      }
+     *    };
+     *  }
+     *
+     * All elements read between the last time this method was called (or since this reader was created, if this method has not been called on this reader) until this method is called will be processed together as a bundle. (An element is considered 'read' if it could be returned by a call to Source.Reader.getCurrent().)
+     *
+     * Once the result of processing those elements and the returned checkpoint have been durably committed, UnboundedSource.CheckpointMark.finalizeCheckpoint() will be called at most once at some later point on the returned UnboundedSource.CheckpointMark object. Checkpoint finalization is best-effort, and checkpoints may not be finalized. If duplicate elements may be produced if checkpoints are not finalized in a timely manner, UnboundedSource.requiresDeduping() should be overridden to return true, and getCurrentRecordId() should be overridden to return unique record IDs.
+     *
+     * A checkpoint will be committed to durable storage only if all all previous checkpoints produced by the same reader have also been committed.
+     *
+     * The returned object should not be modified.
+     *
+     * May not be called before start() has been called.
+     * @return
+     */
+
+    @Override
+    public CheckpointMark getCheckpointMark() {
+        return new UnboundedSource.CheckpointMark() {
+            @Override
+            public void finalizeCheckpoint() throws IOException {}
+        };	}
```

Comment:
New line after `};` ?

Reply:
fixed syntax and indents https://github.com/GoogleCloudPlatform/professional-services/pull/191/commits/05ab1977716987c886d5bb7bb221ea4c72cdc1ee